scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala

```diff
@@ -1215,6 +1215,57 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
   @deprecated("Use readAllAsBytes instead", "0.8.1")
   def readAllBytes(implicit ev: T <:< String): SCollection[Array[Byte]] = readFilesAsBytes
 
+  /**
+   * Pairs each element with the value of the provided [[SideInput]] in the element's window.
+   *
+   * Reify as List:
+   * {{{
+   *  val other: SCollection[Int] = sc.parallelize(Seq(1))
+   *  val coll: SCollection[(Int, Seq[Int])] = sc.parallelize(Seq(1, 2)).reify(other.asListSideInput)
+   * }}}
+   *
+   * Reify as Iterable:
+   * {{{
+   *  val other: SCollection[Int] = sc.parallelize(Seq(1))
+   *  val coll: SCollection[(Int, Iterable[Int])] = sc.parallelize(Seq(1, 2)).reify(other.asIterableSideInput)
+   * }}}
+   *
+   * Reify as Map:
+   * {{{
+   *  val other: SCollection[(Int, Int)] = sc.parallelize(Seq((1, 1)))
+   *  val coll: SCollection[(Int, Map[Int, Int])] = sc.parallelize(Seq(1, 2)).reify(other.asMapSideInput)
+   * }}}
+   *
+   * Reify as Multimap:
+   * {{{
+   *  val other: SCollection[(Int, Int)]  = sc.parallelize(Seq((1, 1)))
+   *  val coll: SCollection[(Int, Map[Int, Iterable[Int]])]  = sc.parallelize(Seq(1, 2)).reify(other.asMultiMapSideInput)
+   * }}}
+   */
+  def reify[U: Coder](side: SideInput[U]): SCollection[(T, U)] = {
```

Comment:
This feels like a case of `def cross[U](that: SCollection[U])` where `that` is a singleton. Is there a better name to improve discovery? Something like `crossSideInput/crossSingleton(that.asSingletonSideInput)` so it's more obvious for tab-completers?

Reply:
I like the idea of having this more obvious ... not sure about the naming though. :thinking:

Reply:
If we rename the other as `reify{Singleton,List,Map,Set}InGlobalWindow`, can call this `reifyWithSideInput`? That also opens other `reify*` options and avoid potential overloading issues.

Reply:
yeah the other one I was thinking, was smth along those lines but with `As``reifyAs{Singleton,List,Map,Set}InGlobalWindow. I think I like `reifyWithSideInput`